From 6c8a3849d600d5f689ace594efb53aef4832bbf6 Mon Sep 17 00:00:00 2001
From: Daniel Borkmann <dborkman@redhat.com>
Date: Tue, 22 Jan 2013 11:06:25 +0100
Subject: [PATCH 32/53] test_1_to_1_nonblock: remove unused iov and malloced
 send buffer
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes:

  test_1_to_1_nonblock.c: In function ‘main’:
  test_1_to_1_nonblock.c:75:13: warning: variable ‘count’ set but not used [-Wunused-but-set-variable]
  test_1_to_1_nonblock.c:72:22: warning: variable ‘iov’ set but not used [-Wunused-but-set-variable]

Also remove count, we probe for received messages anyways.

$ ./test_1_to_1_nonblock
test_1_to_1_nonblock.c  1 PASS : non-blocking accept() - EAGAIN
test_1_to_1_nonblock.c  2 PASS : non-blocking connect() - EINPROGRESS
test_1_to_1_nonblock.c  3 PASS : accept() after a non-blocking connect - SUCCESS
test_1_to_1_nonblock.c  4 PASS : non-blocking recvmsg() - EAGAIN
test_1_to_1_nonblock.c  5 PASS : non-blocking recvmsg() when data is available - SUCCESS

Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
---
 src/func_tests/test_1_to_1_nonblock.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/src/func_tests/test_1_to_1_nonblock.c b/src/func_tests/test_1_to_1_nonblock.c
index bad3a99..1179fb1 100644
--- a/src/func_tests/test_1_to_1_nonblock.c
+++ b/src/func_tests/test_1_to_1_nonblock.c
@@ -69,14 +69,11 @@ main(int argc, char *argv[])
 	struct msghdr outmessage;
 	struct msghdr inmessage;
         char *message = "hello, world!\n";
-        struct iovec iov;
         struct iovec iov_rcv;
 	struct sctp_sndrcvinfo *sinfo;
-        int count;
 	char outcmsg[CMSG_SPACE(sizeof(struct sctp_sndrcvinfo))];
 	struct cmsghdr *cmsg;
         struct iovec out_iov;
-        char * buffer_snd;
 	char * buffer_rcv;
 	char incmsg[CMSG_SPACE(sizeof(sctp_cmsg_data_t))];
 	
@@ -158,8 +155,6 @@ main(int argc, char *argv[])
 	tst_resm(TPASS, "accept() after a non-blocking connect - SUCCESS");
 
 	memset(&outmessage, 0, sizeof(outmessage));
-        buffer_snd = malloc(REALLY_BIG);
-
         outmessage.msg_name = &svr_addr;
         outmessage.msg_namelen = sizeof(svr_addr);
         outmessage.msg_iov = &out_iov;
@@ -176,10 +171,7 @@ main(int argc, char *argv[])
 	sinfo = (struct sctp_sndrcvinfo *)CMSG_DATA(cmsg);
         memset(sinfo, 0x00, sizeof(struct sctp_sndrcvinfo));
 
-	iov.iov_base = buffer_snd;
-        iov.iov_len = REALLY_BIG;
         outmessage.msg_iov->iov_base = message;
-
         outmessage.msg_iov->iov_len = strlen(message) + 1;
 
 	memset(&inmessage, 0, sizeof(inmessage));
@@ -202,7 +194,7 @@ main(int argc, char *argv[])
 
 	tst_resm(TPASS, "non-blocking recvmsg() - EAGAIN");
 
-	count = test_sendmsg(acpt_sk, &outmessage, flag, msg_count);
+	test_sendmsg(acpt_sk, &outmessage, flag, msg_count);
 
 	/* TEST5: recvmsg() should succeed now as data is available. */
 	error = test_recvmsg(sk, &inmessage, flag);
-- 
1.7.11.7

